### PR TITLE
IGNITE-14225: Introduce expectThrows As A Helper Test Method

### PR DESCRIPTION
**Pull request by atris**
This commit introduces expectThrows method which allows tests to capture a block of code and define an
expectation to throw a certain type of exception with a message to print if the block of code does not
do so.

### CODE REVIEW

**Comment by atris** [comment]
@alamar @SammyVimes

**Comment by nizhikov** [comment]
How new method differs from `GridTestUtils#assertThrows`?

**Comment by ibessonov** [comment]
Agree with @nizhikov, wanted to write the same comment.

File: modules/core/src/test/java/org/apache/ignite/testframework/junits/JUnitAssertAware.java [hunk 1]
```diff
@@ -203,4 +208,64 @@ protected static void assertNotSame(Object unexpected, Object actual) {
...
     }
+
+    /** A runnable that can throw any checked exception. */
+    @FunctionalInterface
+    public interface ThrowingRunnable {
+        void run() throws Throwable;
+    }
+
+    /** Checks a specific exception class is thrown by the given runnable, and returns it. */
+    public static <T extends Throwable> T expectThrows(
```
Comment (Mmuzaf) [inline modules/core/src/test/java/org/apache/ignite/testframework/junits/JUnitAssertAware.java:219]: Hello, can you check these methods? Are they the same?

```
org.apache.ignite.testframework.GridTestUtils#assertThrows

org.apache.ignite.testframework.GridTestUtils#assertThrowsWithCause

org.apache.ignite.testframework.GridTestUtils#assertThrowsAnyCause
```

**Comment by atris** [comment]
My bad, for some reason, I did not find those method! Closing this PR.

**Pull request closed without merging**